editor/src/components/editor/component-button.tsx

```diff
@@ -39,39 +46,46 @@ export const ComponentOrInstanceIndicator = betterReactMemo('ComponentOrInstance
     [setIsOpen],
   )
 
-  enum editContext {
-    ELEMENT = 0,
-    COMPONENT_INSTANCE,
-    COMPONENT,
-    COMPONENT_CHILD,
-  }
+  const target = selectedViews[0]
 
-  const [currentEditContext, setCurrentEditContext] = React.useState(editContext.COMPONENT)
+  const isFocused = React.useCallback(() => {
+    if (target != null) {
+      return EP.isFocused(focusedElementPath, target)
+    } else {
+      return false
+    }
+  }, [focusedElementPath, target])
 
-  // TODO replace this with a function that enters focus mode
-  const TODOREPLACEMEcycleEditContext = React.useCallback(() => {
-    const nextEditContext = (currentEditContext + 1) % 4
-    setCurrentEditContext(nextEditContext)
-  }, [currentEditContext])
+  const toggleFocusMode = React.useCallback(() => {
+    dispatch([setFocusedElement(isFocused() ? null : target)])
+  }, [dispatch, isFocused, target])
+
+  const isComponent = React.useCallback(() => {
```

Comment:
same here, it _seems like_ there's no reason why this isn't just a `const isComponent = MetadataUtils.isFocusableComponent(target, metadata)`